lib/ansible/modules/source_control/git.py

```diff
@@ -977,6 +1011,32 @@ def main():
     track_submodules = module.params['track_submodules']
 
     result.update(before=None)
+
+    if archive:
+        try:
+            os.makedirs(dest)
+        except OSError as e:
+            pass
+
+        if archive_file is None:
+            archive_file = "%s/%s.%s" % (dest, version, archive_fmt)
+
+        repo_name = repo.split("/")[-1].replace(".git", "")
+        # Git archive is not supported by all git servers, so
+        # we will first clone and perform git archive from local directory
+        cmd = "%s clone %s %s" %(git_path, repo, repo_name)
+        (rc, out, err) = module.run_command(cmd, cwd=dest)
+        if rc != 0:
+            module.fail_json(msg="Failed to clone source")
```

Comment:
This will fail if the destination exists. Is that the desired behavior? Since it is making a directory, maybe it should check if the directory already exists and if not create it? If it does exist, maybe we should be able to continue onto the archiving step.
In addition, here's a hypothetical scenario: if this archiving fails the first time because the user uses an invalid URL to clone this part of the operation will succeed and the directory will be created. When the URL is fixed and this operation will run a second time it will fail here and not be readily apparent why. I think importing traceback and then failing with `module.fail_json(msg="Failed to clone source", exception=traceback.format_exc())` would be really helpful here for future debugging.